agents/trainer/task.py

```diff
@@ -0,0 +1,319 @@
+  normalize_ranges = True
+  max_length = 1000
+
+  # Network
+  network = agents.scripts.networks.feed_forward_gaussian
+  weight_summaries = dict(
+      all=r'.*', policy=r'.*/policy/.*', value=r'.*/value/.*')
+  policy_layers = 200, 100
+  value_layers = 200, 100
+  init_output_factor = 0.1
+  init_logstd = -1
+  init_std = 0.35
+
+  # Optimization
+  update_every = 60
+  update_epochs = 25
+  optimizer = tf.train.AdamOptimizer
+  learning_rate = 1e-4
+  steps = 3e7  # 30M
+
+  # Losses
+  discount = 0.995
+  kl_target = 1e-2
+  kl_cutoff_factor = 2
+  kl_cutoff_coef = 1000
+  kl_init_penalty = 1
+
+  return locals()
+
+
+def _object_import_from_string(name):
+
+  components = name.split('.')
+  mod = __import__(components[0])
+  for comp in components[1:]:
+    mod = getattr(mod, comp)
+  return mod
+
+
+def _realize_import_attrs(d, filter):
+  for k, v in d.items():
+    if k in filter:
+      # try:
+      imported = _object_import_from_string(v)
+      # except ImportError as e:
+      #     msg = ("Failed to realize import path %s." % v)
+      #     raise e
+      d[k] = imported
+  return d
+
+
+def _get_agents_configuration(hparam_set_name, log_dir=None, is_chief=False):
+  """Load hyperparameter config."""
+  try:
+    # Try to resume training.
+    hparams = agents.scripts.utility.load_config(log_dir)
+  except IOError:
+
+    hparams = hparams_base()
+
+    # --------
+    # Experiment extending base hparams with FLAGS and dynamic import of
+    # network and algorithm.
+    for k, v in FLAGS.__dict__['__flags'].items():
+      hparams[k] = v
+    hparams = _realize_import_attrs(
+        hparams, ["network", "algorithm", "optimizer"])
+    # --------
+
+    hparams = agents.tools.AttrDict(hparams)
+
+    if is_chief:
+      # Write the hyperparameters for this run to a config YAML for posteriority
+      hparams = agents.scripts.utility.save_config(hparams, log_dir)
+
+  pprint.pprint(hparams)
+  return hparams
+
+
+def _dump_dependency_versions():
+  tf.logging.info("Tensorflow version: %s", tf.__version__)
+  tf.logging.info("Tensorflow git version: %s", tf.__git_version__)
+
+  pprint.pprint(sorted(["%s==%s" % (i.key, i.version)
+                        for i in pip.get_installed_distributions()]))
+
+  if FLAGS.run_mode == 'render':
+    os.system("ffmpeg -version")
+
+
+def gcs_upload(local_dir, gcs_out_dir):
+  """Upload the contents of a local directory to a specific GCS path.
+
+  Args:
+    local_dir (str): The local directory containing files to upload.
+    gcs_out_dir (str): The target Google Cloud Storage directory path.
+
+  Raises:
+    ValueError: If `gcs_out_dir` does not start with "gs://".
+
+  """
+
+  # Get a list of all files in the local_dir
+  local_files = [f for f in os.listdir(
+      local_dir) if os.path.isfile(os.path.join(local_dir, f))]
+  tf.logging.info("Preparing local files for upload:\n %s" % local_files)
+
+  # Initialize the GCS API client
+  storage_client = storage.Client()
```

Comment:
Would it be possible to use [`tf.gfile`](https://www.tensorflow.org/api_docs/python/tf/gfile/GFile) here instead?

(Even if the workers are writing to local disk, you can use [`tf.gfile.Copy`](https://www.tensorflow.org/api_docs/python/tf/gfile/Copy).)